File: dotnet/cafdemalihapa/Directories/Hosting/Feature/FeatureDirectory.cs
namespace cafdemalihapa.Directories.Hosting.Feature
{
    public class FeatureDirectory(
            HostingDirectory hostingDirectory
        )
    {
        public void Create()
        {
            var featureDirectoryPath = GetPath();
            Directory.CreateDirectory(featureDirectoryPath);
        }

        public string GetPath()
        {
            var featureName = CommandLineArgs.GetByKey("--feature-name");
            var hostingDirectoryName = hostingDirectory.GetPath();
            var featureDirectoryPath = Path.Combine(hostingDirectoryName, featureName);
            return featureDirectoryPath;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/FeatureDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature

open System.IO
open cafdemwimu.console
open cafdemwimu.console.Directories.Hosting

[<AbstractClass; Sealed>]
type FeatureDirectory private () =
    static member Create() =
        let featureDirectoryPath = FeatureDirectory.GetPath()
        Directory.CreateDirectory(featureDirectoryPath) |> ignore

    static member GetPath() =
        let featureName = CommandLineArgs.GetByKey("--feature-name")
        let hostingDirectoryName = HostingDirectory.GetPath()
        let featureDirectoryPath = Path.Combine(hostingDirectoryName, featureName)
        featureDirectoryPath
